### PR TITLE
more return types

### LINKED ISSUE #191: Support more return types in predict
Currently, the predict function returns only categories.

As a user I want to set the return type in the predict function, so I can get a label or raw probabilities instead of just the category.

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?
Fixes #191 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/model.py [hunk 1]
```diff
@@ -123,6 +127,8 @@ def predict(
         skip_collate_fn: bool = False,
         dataloader_idx: Optional[int] = None,
         data_pipeline: Optional[DataPipeline] = None,
+        return_type: str = None,
+        labels: Optional[List[str]] = None,
```
Comment (Borda) [inline flash/core/model.py:131]: is it updated also in docs?

Comment (aniketmaurya) [reply flash/core/model.py:133]: @Borda are docs created automatically with help of docstrings?

Comment (Borda) [reply flash/core/model.py:133]: yes, types and default are also in docs

**Comment by Borda** [comment]
@aniketmaurya mind check collision :]

**Comment by aniketmaurya** [comment]
Sure @Borda

**Comment by carmocca** [comment]
Note that this should be handled inside a `Postprocess`, not in the `predict_step`

**Comment by aniketmaurya** [comment]
> Note that this should be handled inside a `Postprocess`, not in the `predict_step`

@carmocca are you suggesting to put `postprocess` in the `data_pipeline`? Or pass the arguments in `predict` func and a separate postprocess func should take care of it? Because I see that data_pipeline has to be passed by user as argument and it will not be convenient for users to create data_pipeline for printing classes/raw probs.

**Comment by carmocca** [comment]
DataPipeline was completely revamped lately. It now includes Preprocess and Postprocess objects.

These return conversions are now expected to be encapsulated by the postprocess part. What is your use case exactly?

For example, this is an example setting a custom Preprocess

https://github.com/PyTorchLightning/lightning-flash/blob/master/flash/core/classification.py

cc: @tchaton

File: flash/core/model.py [hunk 2]
```diff
@@ -55,6 +56,11 @@ class Task(LightningModule):
         learning_rate: Learning rate to use for training, defaults to `5e-5`
     """
 
+    class Return(Enum):
```
Comment (tchaton) [inline flash/core/model.py:59]: I like the idea, but the PR is outdate. Do you mind merging with master. Also, this logic should be handled by the postprocess object.

Comment (aniketmaurya) [reply flash/core/model.py:59]: hi @tchaton , as @carmocca pointed out that DataPipeline is completely revamped and codes were dependent on prev flow. I will create the logic again with postprocess object and update the PR.

**Comment by ethanwharris** [comment]
Hi @aniketmaurya, great work on this PR :smiley: I've now added multiple return types in #229 (with some inspiration from your work) so closing this.

Please feel free to provide your thoughts on #229
